### PR TITLE
Use standard c++ exceptions

### PR DESCRIPTION
**Pull request by CendioOssman**
Clean up our exception handling by using the standard exception classes provided by C++.

### CODE REVIEW

File: common/rfb/Exception.h [hunk 1]
```diff
@@ -18,17 +18,22 @@
...
 namespace rfb {
-  typedef rdr::Exception Exception;
-  struct AuthFailureException : public Exception {
+  class protocol_error : public std::runtime_error {
+  public:
+    protocol_error(const char* what_arg) : std::runtime_error(what_arg) {}
+    protocol_error(const std::string& what_arg) : std::runtime_error(what_arg) {}
+  };
+
+  struct AuthFailureException : public std::runtime_error {
```
Comment (samhed) [inline common/rfb/Exception.h]: Did the `AuthFailureException` and `AuthCancelledException` end up in the wrong commit here?

File: common/rfb/CSecurityTLS.cxx [hunk 2]
```diff
@@ -201,18 +201,16 @@ void CSecurityTLS::setParam()
     char *prio;
     const char *err;
 
-    prio = (char*)malloc(strlen(Security::GnuTLSPriority) +
-                         strlen(kx_anon_priority) + 1);
-    if (prio == nullptr)
-      throw Exception("Not enough memory for GnuTLS priority string");
+    prio = new char[strlen(Security::GnuTLSPriority) +
+                    strlen(kx_anon_priority) + 1];
```
Comment (samhed) [inline common/rfb/CSecurityTLS.cxx:205]: Sweet!

**Review by samhed — APPROVED** [review]
Looks great!

File: common/rfb/Exception.h (hunk 1 above)
Comment (CendioOssman) [reply common/rfb/Exception.h]: I split things a bit more now to make it more clear. Please have a look.

**Review by samhed — APPROVED** [review]
LGTM!

**Pull request merged**